build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,350 @@
+        sh "cp xa-build-status*.zip ${packageDir}"
+
+    } catch (error) {
+        echo "processAndStageBuildResults: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+def publishPackages() {
+    def status = 0
+    try {
+        def packageDir = "${env.WORKSPACE}/package"
+        sh "cd ${packageDir}"
+
+        echo "Publishing to Azure storage account: https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/7b4817ae-218f-464a-bab1-a9df2d99e1e5/resourceGroups/Group/providers/Microsoft.ClassicStorage/storageAccounts/xamjenkinsartifact/overview"
+        step([$class: 'WAStoragePublisher', allowAnonymousAccess: false, cleanUpContainer: false, cntPubAccess: false, containerName: "${env.ContainerName}", doNotFailIfArchivingReturnsNothing: false,
+            doNotUploadIndividualFiles: false, doNotWaitForPreviousBuild: false, excludeFilesPath: '',
+            filesPath: '*.*', storageAccName: "${env.StorageAccountName}", storageCredentialId: "${env.StorageCredentialId}",
+            uploadArtifactsOnlyIfSuccessful: false, usploadZips: false, virtualPath: "${env.StorageVirtualPath}", storageType: 'blobstorage'
+        ])
+
+    } catch (error) {
+        echo "publishPackages: Unexpected error: ${error}"
+        status = 1
+    }
+
+    return status
+}
+
+pipeline {
+    
+    agent {
+        label 'xamarin-android-master-pipeline-eval'
+    }
+
+    options {
+        timestamps()
+    }
+
+    stages {
+        stage('clean') {
+            steps {
+                dir('package') {
+                    deleteDir()
+                }
+
+                dir('package.tests') {
+                    deleteDir()
+                }
+
+                dir('xamarin-android@tmp') {
+                    deleteDir()
+                }
+
+                dir('xamarin-android') {
+                    echo "Cleaning build status files "
+
+                    sh "rm -f xa-build-status*.zip"
+
+                    sh '''
+                    for dir in `ls -d xa-build-status*`; do
+                        [ -d "$dir" ] || continue
+                        echo "Cleaning directory '${dir}''"
+                        rm -rf "$dir"
+                    done
+                    '''
+                }
+            }   
+        }
+    
+        stage('prepare deps') {
+            options { timeout(time: 30, unit: 'MINUTES') }     // Typically takes less than 2 minutes
+            steps {
+                dir("$XADir") {
+                    sh "make prepare-deps CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                }
+            }
+        }
+        
+        stage('build') {
+            options { timeout(time: 3, unit: 'HOURS') }     // Typically takes less than 2 hours
+            steps {
+                dir("$XADir") {
+                    sh "make jenkins CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
```

Comment:
Yup, it can be done. I've applied a pattern similar to what the monodroid pipeline build uses; basically, conditional logic based on JOB_BASE_NAME looking for any job that begins with 'xamarin-android-pr'.  Implemented with this commit: https://github.com/xamarin/xamarin-android/commit/0a0b385650c86dfbc296a89e27239341f1fb4a43